sources/kms/key-policy.go

```diff
@@ -0,0 +1,104 @@
+package kms
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/service/kms"
+	"github.com/overmindtech/aws-source/sources"
+	"github.com/overmindtech/sdp-go"
+)
+
+type keyPolicyClient interface {
+	GetKeyPolicy(ctx context.Context, params *kms.GetKeyPolicyInput, optFns ...func(*kms.Options)) (*kms.GetKeyPolicyOutput, error)
+	ListKeyPolicies(ctx context.Context, params *kms.ListKeyPoliciesInput, optFns ...func(*kms.Options)) (*kms.ListKeyPoliciesOutput, error)
+}
+
+func getKeyPolicyFunc(ctx context.Context, client keyPolicyClient, scope string, input *kms.GetKeyPolicyInput) (*sdp.Item, error) {
+	output, err := client.GetKeyPolicy(ctx, input)
+	if err != nil {
+		return nil, err
+	}
+
+	if output.Policy == nil {
+		return nil, &sdp.QueryError{
+			ErrorType:   sdp.QueryError_NOTFOUND,
+			ErrorString: "get key policy response was nil",
+		}
+	}
+
+	attributes, err := sources.ToAttributesCase(output)
+	if err != nil {
+		return nil, err
+	}
+
+	err = attributes.Set("keyId", *input.KeyId)
+	if err != nil {
+		return nil, err
+	}
+
+	item := &sdp.Item{
+		Type:            "kms-key-policy",
+		UniqueAttribute: "keyId",
+		Attributes:      attributes,
+		Scope:           scope,
+	}
+
+	// +overmind:link kms-key
+	item.LinkedItemQueries = append(item.LinkedItemQueries, &sdp.LinkedItemQuery{
+		Query: &sdp.Query{
+			Type:   "kms-key",
+			Method: sdp.QueryMethod_GET,
+			Query:  *input.KeyId,
+			Scope:  scope,
+		},
+		BlastPropagation: &sdp.BlastPropagation{
+			// These are tightly coupled
+			In:  true,
+			Out: true,
+		},
+	})
+
+	return item, nil
+}
+
+//go:generate docgen ../../docs-data
+// +overmind:type kms-key-policy
+// +overmind:descriptiveType KMS Key Policy
+// +overmind:get Get a KMS key policy by its Key ID
+// +overmind:search Search KMS key policies by Key ID
+// +overmind:group AWS
+// +overmind:terraform:queryMap aws_kms_key_policy.key_id
+
+func NewKeyPolicySource(client keyPolicyClient, accountID string, region string) *sources.AlwaysGetSource[*kms.ListKeyPoliciesInput, *kms.ListKeyPoliciesOutput, *kms.GetKeyPolicyInput, *kms.GetKeyPolicyOutput, keyPolicyClient, *kms.Options] {
+	return &sources.AlwaysGetSource[*kms.ListKeyPoliciesInput, *kms.ListKeyPoliciesOutput, *kms.GetKeyPolicyInput, *kms.GetKeyPolicyOutput, keyPolicyClient, *kms.Options]{
+		ItemType:    "kms-key-policy",
+		Client:      client,
+		AccountID:   accountID,
+		Region:      region,
+		DisableList: true, // This source only supports listing by Key ID
+		SearchInputMapper: func(scope, query string) (*kms.ListKeyPoliciesInput, error) {
+			return &kms.ListKeyPoliciesInput{
+				KeyId: &query,
```

Comment:
If you can search by Key ID, can we also add a LinkedItemQuery to the `kms-key` resource that always does a `SEARCH` for these?

Reply:
So, two links for the kms key, right?
One for get and one for search?

Reply:
Should I do the same for grant?